src/common/base/Status.h

```diff
@@ -60,6 +60,10 @@ class Status final {
         return code() == rhs.code();
     }
 
+    bool operator!=(const Status &rhs) const {
```

Comment:
Why do we need this? My understanding is the compiler will convert ```!=``` to ```! ==()```

Reply:
No.  The compiler will not do that.